File: CoreJavaDay2Assignment/Truck.java
package com.sonata;

public class Truck extends Car {
	int weight;
	Truck(int speed,double regularPrice,String color,int weight)
	{
		super(speed,regularPrice,color);
		this.weight=weight;
	}
	public double getSalePrice()
	{
		if(weight>2000)
		{
			return super.getSalePrice()-(0.1*super.getSalePrice());
		}
		else
		{
			return super.getSalePrice()-(0.2*super.getSalePrice());
		}
	}
	public static void main(String args [])
	{
		Truck t=new Truck(200,1400000,"black",2500);
		System.out.println(t.getSalePrice());
	}
	
	

}

File: Typescript_Assignment/Truck.ts
import { Car } from "./Car"

class Truck extends Car {

    weight: number;
    constructor(s: number, r: number, c: string, w: number) {
        super(s, r, c);
        this.weight = w;
    }

    getSalePrice(): number {
        if (this.weight > 2000) {
            return super.getSalePrice() - (0.1 * super.getSalePrice());
        }
        else {
            return super.getSalePrice() - (0.2 * super.getSalePrice());

        }

    }
}

var truck = new Truck(200, 2500000, "White", 3000);
console.log(truck.getSalePrice());